libvast/vast/config.hpp.in

```diff
@@ -32,9 +32,19 @@ extern const char* build_tree_hash;
 
 } // namespace vast::version
 
-#define VAST_DATADIR "@CMAKE_INSTALL_FULL_DATADIR@/vast"
-#define VAST_CONFIGDIR "@CMAKE_INSTALL_FULL_SYSCONFDIR@/vast"
-#define VAST_PLUGINDIR "@CMAKE_INSTALL_FULL_LIBDIR@/vast/plugins"
+// NOTE: Do not use these directly, but rather use
+// `vast::detail::install_*dir()` functions that correctly use the rpath for
+// non-relocatable builds.
+
+#if VAST_ENABLE_RELOCATABLE_INSTALLATIONS
+#  define VAST_DATADIR "@CMAKE_INSTALL_DATADIR@/vast"
+#  define VAST_CONFIGDIR "@CMAKE_INSTALL_SYSCONFDIR@/vast"
+#  define VAST_PLUGINDIR "@CMAKE_INSTALL_LIBDIR@/vast/plugins"
+#else
+#  define VAST_FULL_DATADIR "@CMAKE_INSTALL_FULL_DATADIR@/vast"
+#  define VAST_FULL_CONFIGDIR "@CMAKE_INSTALL_FULL_SYSCONFDIR@/vast"
+#  define VAST_FULL_PLUGINDIR "@CMAKE_INSTALL_FULL_LIBDIR@/vast/plugins"
+#endif
```

Comment:
To avoid accidental misuse because of autocompletion or something similar, I think we should move these and the implementation of the `installdirs` functions into `config.cpp.in`.

Reply:
I opted to configure the `installdirs.cpp` file instead to avoid having to include `<filesystem>` in `config.hpp`.